integration_tests/src/main/python/data_gen.py

```diff
@@ -681,6 +713,17 @@ def skip_if_not_utc():
     if (not is_tz_utc()):
         skip_unless_precommit_tests('The java system time zone is not set to UTC')
 
+@lru_cache(maxsize=128)
```

Comment:
consider
```suggestion
@lru_cache(maxsize=128, typed=True)
```

Reply:
Thanks for the review! I don't think the `typed` here can affect the behaviour of `hash_object`, because `@lru_cache` checks if an object is in the cache with the `__eq__` method, and I just use the hash value to check for equality.

Reply:
having type as an input to cache's hash key makes unexpected side effects from another [thread](https://github.com/NVIDIA/spark-rapids/pull/8441#discussion_r1213172989) go away.